File: src/main/java/com/zhimian/model/entity/KnowledgeDocument.java
package com.zhimian.model.entity;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class KnowledgeDocument {
    private Long id;
    private String title;
    private String originalFilename;
    private String sourceType;
    private String content;
    private String contentHash;
    private Integer status;
    private Integer chunkCount;
    private String embeddingModel;
    private Integer vectorDimension;
    private String errorMessage;
    private Long createdBy;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    private Integer isDeleted;
}

File: frontend/src/types/knowledge-document.ts
export type KnowledgeDocumentStatus = 0 | 1 | 2 | 3 | 4

export interface KnowledgeDocumentImportPayload {
  title: string
  originalFilename: string
  sourceType: 'MARKDOWN' | 'TEXT'
  content: string
}

export interface KnowledgeDocumentStatusVO {
  id: number
  title: string
  status: KnowledgeDocumentStatus
  chunkCount: number
  embeddingModel: string | null
  vectorDimension: number | null
  errorMessage: string | null
  updateTime: string | null
}

export interface TrackedKnowledgeDocument {
  id: number
  submittedAt: string
}
